OpenSim/Simulation/Wrap/WrapCylinderObst.cpp

```diff
@@ -110,76 +80,55 @@ void WrapCylinderObst::initCircleWrapPts()
 /**
 * Connect properties to local pointers.
 */
-void WrapCylinderObst::setupProperties()
+void WrapCylinderObst::constructProperties()
 {
-    _radiusProp.setName("radius");
-    _radiusProp.setValue(-1.0);
-    _propertySet.append(&_radiusProp);
-
-    _wrapDirectionNameProp.setName("wrapDirection");
-    _wrapDirectionNameProp.setValue("Unassigned");
-    _propertySet.append(&_wrapDirectionNameProp);
-
-    _lengthProp.setName("length");
-    _lengthProp.setValue(1.0);
-    _propertySet.append(&_lengthProp);
+    constructProperty_radius(-1.0);
+    constructProperty_wrapDirection("Unassigned");
+    m_wrapDirection = righthand;
+    constructProperty_length(1.0);
```

Comment:
Thanks for doing this. I know we're just reusing initialization values from old code but it may make sense to initialize the object to valid parameters with positive radius and valid wrapDirection so that users can create a valid instance with minimal code then use scripting or GUI to modify/update as needed.